source/objects/RenderTarget.cpp

```diff
@@ -209,6 +214,119 @@ Pixel RenderTarget::SampleBilinear(float u, float v, uint8_t mip) const
 	};
 }
 
+bool RenderTarget::Load(const char* filename) {
+	if (!IsValid()) return false;
+
+	int width = 0;
+	int height = 0;
+	int channels = CHANNELS[static_cast<uint8_t>(GetFormat())];
+	
+	size_t stride = STRIDES[static_cast<uint8_t>(GetFormat())];
+
+	if (stride == sizeof(uint8_t)) {
+		// Load it as an unsigned char array
+		stbi_uc* imageBuffer = stbi_load(filename, &width, &height, nullptr, channels);
+		if (!imageBuffer) {
+			return false;
+		}
+
+		bool resized = Resize(width, height);
+		if (!resized) {
+			stbi_image_free(imageBuffer);
+			return false;
+		}
+
+		memcpy(GetRawData(), imageBuffer, stride * width * height * channels);
+		GenerateMIPs();
+		stbi_image_free(imageBuffer);
+
+		return true;
+	}
+	else if (stride == sizeof(float)) {
+		// Load it as a float array
+		float* imageBuffer = stbi_loadf(filename, &width, &height, nullptr, channels);
+		if (!imageBuffer) {
+			return false;
+		}
+
+		bool resized = Resize(width, height);
+		if (!resized) {
+			stbi_image_free(imageBuffer);
+			return false;
+		}
+
+		memcpy(GetRawData(), imageBuffer, stride * width * height * channels);
+		GenerateMIPs();
+		stbi_image_free(imageBuffer);
+
+		return true;
+	}
+
+	return false; // wtf? invalid stride?
+}
+
+bool RenderTarget::Write(const char* filename, uint8_t mips) {
+	uint8_t* imageData = GetRawData(mips);
+
+	size_t stride = STRIDES[static_cast<uint8_t>(GetFormat())];
+	uint8_t channels = CHANNELS[static_cast<uint8_t>(GetFormat())];
+
+	const bool isFloat = stride == sizeof(float);
+
+	const char* extension = isFloat ? ".hdr" : ".png";
+
+	fs::path workingDir = fs::current_path();
+	workingDir += "/garrysmod/data/";
+	workingDir = workingDir.make_preferred(); // Convert to forwardslashes on Windows and backslashes on POSIX.
+
+	fs::path filenamePath(filename);
+	filenamePath = filenamePath.make_preferred();
+
+	// Do not allow going up the file tree
+	const char* transformedFilename = filenamePath.string().c_str();
```

Comment:
You should just get the abs path with `filesystem` and check the root vs the `data/vistrace` folder